vip-scanner/checks/VIPRestrictedVariablesCheck.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+class VIPRestrictedVariablesCheck extends BaseCheck
+{
+	function check( $files ) {
+		$result = true;
+
+		$checks = array(
+			'((?<![\\\'\"])\$\$[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*?(?![\\\'\"])|(?<![\\\'\"])\$\{(?:.*)[\}](?![\\\'\"]))' => array( "level" => "Warning", "note" => "Possible variable variables" ),
```

Comment:
> Will this regex match variable variables using bracket notation?

It should match variable variables with bracket notation: `\$\{(?:.*)[\}]`

The one place where this might be a problem is if someone is trying to use a variable variable with bracket notation inside of a double quoted string literal.  Even though the variable variable will not expand, the expression inside the [complex syntax](http://php.net/manual/en/language.types.string.php#language.types.string.parsing.complex) will expand.  This could cause more problems than even having a regular variable variable since the output of the expression may have unexpected results the author never intended.